### PR TITLE
fix: count for error response and network errors

### PR DESCRIPTION
**Pull request by metcoder95**
## Status

- [x] I have read and agreed to the [Developer's Certificate of Origin][cert]
- [x] Tested
- [ ] Benchmarked (**optional**)
- [x] Documented
- [x] Review ready
- [ ] In review
- [ ] Merge ready

[cert]: https://github.com/nodejs/undici/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by mcollina — CHANGES_REQUESTED** [review]
Thanks for opening a PR! Can you please add a unit test?

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
there is a conflict now, can you fix it?

File: lib/handler/retry-handler.js [hunk 1]
```diff
@@ -159,9 +158,7 @@ class RetryHandler {
     const retryTimeout =
       retryAfterHeader > 0
         ? Math.min(retryAfterHeader, maxTimeout)
-        : Math.min(currentTimeout * timeoutFactor ** counter, maxTimeout)
-
-    state.currentTimeout = retryTimeout
+        : Math.min(minTimeout * timeoutFactor ** counter, maxTimeout)
```
Comment (acommodari) [inline lib/handler/retry-handler.js:161]: if counter starts at 1 should this not be (counter - 1)? This way the first retry uses the `minTimeout` value?

Comment (acommodari) [reply lib/handler/retry-handler.js:161]: right now it would start at double the original minTimeout

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**